src/printer.js

```diff
@@ -1729,6 +1729,31 @@ function genericPrintNoParens(path, options, print, args) {
     case "TemplateElement":
       return join(literalline, n.value.raw.split(/\r?\n/g));
     case "TemplateLiteral": {
+      /* Subtree parsing PoC */
+      const parent = path.getParentNode();
+      const parentParent = path.getParentNode(1);
+      const isStyledJsx =
+        n.quasis &&
+        n.quasis.length === 1 &&
+        parent.type === "JSXExpressionContainer" &&
+        parentParent.type === "JSXElement" &&
+        parentParent.openingElement.attributes.some(
+          attribute => attribute.name.name === "jsx"
+        );
+
+      if (isStyledJsx) {
+        const parseCss = require("./parser-postcss");
+
+        const text = n.quasis[0].value.raw;
```

Comment:
yes I was just going to say that it won't work in all the use cases but I don't think that they are that many though. I have to fix this in styled-jsx so I will let you know if I come up with something :)